src/shared/lib/LazyMobXCache.ts

```diff
@@ -0,0 +1,171 @@
+import Immutable from "seamless-immutable"; // need to use immutables so mobX can observe the cache shallowly
+import accumulatingDebounce from "./accumulatingDebounce";
+import {observable, action} from "mobx";
+import {AccumulatingDebouncedFunction} from "./accumulatingDebounce";
+
+export type CacheData<T> = {
```

Comment:
i'm not totally sure!  I think they are for disambiguating very common names, like Cache for example.  Not a big deal.  I will just merge this and we can revisit namespaces later.